common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -352,6 +362,42 @@ protected void validateTokenRequest(MicrosoftStsTokenRequest request) {
         //TODO implement
     }
 
+    @Override
+    protected HttpResponse performTokenRequest(final MicrosoftStsTokenRequest request) throws IOException, ClientException {
+        final String methodName = ":performTokenRequest";
+        HttpResponse response =  super.performTokenRequest(request);
```

Comment:
Nit : final

Reply:
cannot be final. As the http response will be updated when after resolving the pkeyauth challenge.